logstash-core/src/main/java/org/logstash/config/ir/compiler/EventCondition.java

```diff
@@ -467,35 +468,40 @@ private static boolean valueIsTruthy(Object object) {
                     !Boolean.toString(false).equals(other);
         }
 
+        private static RubyRegexp newRegexp(String pattern) {
+            final Ruby runtime = RubyUtil.RUBY;
+            return RubyRegexp.newRegexpFromStr(runtime, runtime.newString(pattern), 0);
+        }
+
+        private static boolean matches(RubyString str, RubyRegexp regexp) {
+            return regexp.match_p(RubyUtil.RUBY.getCurrentContext(), str).isTrue(); // match? returns true/false
```

Comment:
newRegexp and matches have different style decisions on extracting the runtime into a final variable vs inlining it, any reason to use one vs the other?

Reply:
in one case its used once while the other needs the runtime twice - a left over from the [code that was there before](https://github.com/elastic/logstash/pull/11653/files#diff-5c70c3320810f79e79767590b114ad79L494)